sprint_boundary_iterator.rb

```diff
@@ -36,19 +36,30 @@ def next_boundary
     elsif @last_number.nil?
       @last_number = FIRST_AUTOCOMPUTED_NUMBER
       @last_range  = FIRST_AUTOCOMPUTED_RANGE
+    elsif @last_number < 151
+      @last_range   = compute_next_range_before_152(@last_range)
+      @last_number += 1
     else
-      @last_range   = compute_next_range(@last_range)
+      @last_range   = compute_next_range_152_and_after(@last_range)
       @last_number += 1
     end
 
     [@last_number, @last_range]
   end
 
-  def compute_next_range(current)
+  def compute_next_range_before_152(current)
     date = current.end + 2.weeks
     while (date.month == 12 && (22..31).cover?(date.day)) || (date.month == 1 && (1..4).cover?(date.day))
       date += 1.weeks
     end
-    current = (current.end + 1.day)..date
+    (current.end + 1.day)..date
+  end
+
+  def compute_next_range_152_and_after(current)
+    date = current.end + 2.weeks
+    while (date.month == 12 && (21..31).cover?(date.day)) || (date.month == 1 && (1..3).cover?(date.day))
```

Comment:
Basically, the change is in the numbers:

```diff
-     while (date.month == 12 && (22..31).cover?(date.day)) || (date.month == 1 && (1..4).cover?(date.day))
+     while (date.month == 12 && (21..31).cover?(date.day)) || (date.month == 1 && (1..3).cover?(date.day))
```

I thought it was cleaner to copy the method that try to dry it up, because if we ever do this again it may not be so clean.